manifests/example/cloner-datavolume.yaml

```diff
@@ -0,0 +1,15 @@
+apiVersion: cdi.kubevirt.io/v1alpha1
+kind: DataVolume
+metadata:
+  name: cloner-datavolume
+spec:
+  source:
+    pvc:
+      name: "source-ns/my-data-volume"
```

Comment:
Maybe we should consider splitting the pvc's namespace and name into separate fields here. We can construct the <namespace>/<name> string when setting annotation in the controller.

This would let 'namespace' be optional. When no namespace is set, we'd just default to the namespace of the DataVolume object.

thoughts?

Reply:
I think @davidvossel suggestion regarding the namespace is a great idea FWIW

Reply:
@davidvossel I changed the cloner-datavolume to have the namespace filed along with the name as you recommended. k8s just ignore this or any other field except the name. When I do '-o yaml' I can see only the name field under the pvc. 
Is there a place where er define to k8s how the CRD should look like? Is  there another place I need to change?
Note: this is before I let the controller create the PVC from the cloner-datavolume. This is just the beginning when the cloner-datavolume is created, but with missing fields.

Reply:
The datavolume spec you set is the one that you'll get back.  If you set only the name of the PVC, then you'll only get the name back if you look at the yaml. If you set both the name and namespace, you should see both with 'kubectl get datavolume <dv-name> -o yaml'

Reply:
This is what I thought, but I don't.
Cloner-datavolume.yaml
apiVersion: cdi.kubevirt.io/v1alpha1
kind: DataVolume
metadata:
  name: cloner77-datavolume
spec:
  source:
    pvc:
      namepace: "tzvika"
      name: "source-ns/my-data-volume"
  pvc:
    #storageClassName: "hostpath"
    accessModes:
      - ReadWriteMany
    resources:
      requests:
        storage: 500Mi

$cluster/kubectl.sh get datavolume cloner77-datavolume -n target-ns -o yaml
apiVersion: cdi.kubevirt.io/v1alpha1
kind: DataVolume
metadata:
  clusterName: ""
  creationTimestamp: 2018-08-02T14:23:04Z
  generation: 1
  name: cloner77-datavolume
  namespace: target-ns
  resourceVersion: "4584"
  selfLink: /apis/cdi.kubevirt.io/v1alpha1/namespaces/target-ns/datavolumes/cloner77-datavolume
  uid: 9197ce27-965f-11e8-843a-525500d15501
spec:
  pvc:
    accessModes:
    - ReadWriteMany
    resources:
      requests:
        storage: 500Mi
  source:
    pvc:
      name: source-ns/my-data-volume
status:
  phase: Pending

Reply:
we cannot change the name of the fields. right? i.e if you change the field 'name' to' myname', you will not see it in the -o yaml. There must be somewhere definition of the fields you define in this CRD.

Reply:
Sorry. Had a typo. I'm good

Reply:
@zvikorn so you figured it out? Everything should work.